brute_force.cpp

```diff
@@ -1,18 +1,14 @@
 #include "brute_force.hpp"
-
-// Get arr[idx], where arr is the compact form of the gene sequence.
-static inline char get(const char *arr, int idx) {
-    return (arr[idx>>2] >> ((idx & 0x3) << 1)) & 0x3;
-}
+#include "char_compress.hpp"
 
 int brute_force_search(
-    const char *text, int text_length, const char *pattern, int pattern_length,
-    int *output, int max_output_cnt, int *fail
+    const char *text, int text_length, const char *pattern, int16_t pattern_length,
+    int16_t *output, int max_output_cnt, int16_t *fail
```

Comment:
If we are storing indices in text data in `*output`, will we not overflow if we use int16_t?

Reply:
You're right. Changed to int.